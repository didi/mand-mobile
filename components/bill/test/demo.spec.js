import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import {renderToString} from '@vue/server-test-utils'

describe('Bill', () => {
  test(`Basic`, () => {
    expect(renderToString(Demo0)).toMatchSnapshot()
  })
  test(`Using slots`, () => {
    expect(renderToString(Demo1)).toMatchSnapshot()
  })
})
