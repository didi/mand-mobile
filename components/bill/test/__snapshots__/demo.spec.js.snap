// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill Basic 1`] = `
<div class="md-example-child md-example-child-bill-0">
  <div class="md-water-mark md-bill">
    <div class="water-mark-container">
      <header class="md-bill-header">
        <h4 class="md-bill-title">借款电子票据</h4>
        <div class="md-bill-no">NO.12345689</div>
      </header>
      <div class="md-bill-neck"><span></span></div>
      <div class="md-bill-content">
        <div class="md-bill-detail">
          <div class="md-detail-item">
            <div class="md-detail-title">借款金额</div>
            <div class="md-detail-content">
              ¥30,000
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">收款账户</div>
            <div class="md-detail-content"><svg class="md-icon icon-svg md-icon-bank-zs md" style="fill:;margin-right:10px;">
                <use xlink:href="#bank-zs"></use>
              </svg>招商银行(尾号xxxx)
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">借款期数</div>
            <div class="md-detail-content">12期</div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">正常还款总息</div>
            <div class="md-detail-content">
              ¥140.50
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">还款</div>
            <div class="md-detail-content">
              <div class="md-tag" style="margin-right:4px;">
                <div class="default size-small shape-fillet type-fill font-weight-normal" style="background:#858B9C;">首次</div>
              </div>
              ¥404.50 (9月22日)
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
    <div class="water-mark-list">
      <div class="water-mark-list-wrapper" style="opacity:0.1;transform:rotate(-30deg);"><canvas class="water-mark-canvas"></canvas></div>
    </div>
  </div>
</div>
`;

exports[`Bill Using slots 1`] = `
<div class="md-example-child md-example-child-bill-1">
  <div class="md-water-mark md-bill">
    <div class="water-mark-container">
      <header class="md-bill-header">
        <div class="header-slot">
          <h1 class="title">借款单据</h1>
          <p class="desc">仔细阅读，关注您的利益、义务</p>
        </div>
      </header>
      <div class="md-bill-neck"><span></span></div>
      <div class="md-bill-content">
        <div class="md-bill-detail">
          <div class="md-detail-item">
            <div class="md-detail-title">借款金额</div>
            <div class="md-detail-content">
              ¥30,000
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">收款账户</div>
            <div class="md-detail-content"><svg class="md-icon icon-svg md-icon-bank-zs md" style="fill:;margin-right:10px;">
                <use xlink:href="#bank-zs"></use>
              </svg>招商银行(尾号xxxx)
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">借款期数</div>
            <div class="md-detail-content">12期</div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">正常还款总息</div>
            <div class="md-detail-content">
              ¥140.50
            </div>
          </div>
          <div class="md-detail-item">
            <div class="md-detail-title">还款</div>
            <div class="md-detail-content">
              <div class="md-tag" style="margin-right:4px;">
                <div class="default size-small shape-fillet type-fill font-weight-normal" style="background:#858B9C;">首次</div>
              </div>
              ¥404.50 (9月22日)
            </div>
          </div>
        </div>
        <footer class="md-bill-footer">
          <div class="footer-slot">
            1 账单生成后显示在滴水贷首页，请按时还款避免逾期。 <br>
            2 整笔账单还款完成后，额度将恢复。暂不支持额度实时恢复。<br>
            3 还款日将自动扣款，扣款顺序优先余额，其次还款账户：工商银行 (尾号xxxx)。
          </div>
        </footer>
      </div>
    </div>
    <div class="water-mark-list">
      <div class="water-mark-list-wrapper" style="opacity:0.1;transform:rotate(-30deg);">
        <ul class="water-mark-line" style="margin-bottom:20vw;">
          <li class="water-mark-item" style="margin-right:20vw;"></li>
          <li class="water-mark-item" style="margin-right:20vw;"></li>
        </ul>
        <ul class="water-mark-line" style="margin-bottom:20vw;">
          <li class="water-mark-item" style="margin-left:20vw;"></li>
          <li class="water-mark-item" style="margin-left:20vw;"></li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
